tests/johnsnowlabs/test_jsl_api.py

```diff
@@ -0,0 +1,782 @@
+import json
+import os
+from pathlib import Path
+from unittest import mock
+
+import numpy as np
+import pandas as pd
+import pyspark
+import pytest
+import yaml
+from johnsnowlabs import nlp
+from packaging.version import Version
+
+import mlflow
+import mlflow.johnsnowlabs
+import mlflow.tracking
+import mlflow.utils.file_utils
+from mlflow import pyfunc
+from mlflow.environment_variables import MLFLOW_DFS_TMP
+from mlflow.johnsnowlabs import _add_code_from_conf_to_system_path
+from mlflow.models import Model, infer_signature, build_docker
+from mlflow.models.utils import _read_example
+from mlflow.pyfunc import spark_udf
+from mlflow.store.artifact.s3_artifact_repo import S3ArtifactRepository
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.utils.file_utils import TempDir
+from mlflow.utils.model_utils import _get_flavor_configuration
+from tests.helper_functions import (
+    _compare_conda_env_requirements,
+    _assert_pip_requirements,
+    _compare_logged_code_paths,
+    _mlflow_major_version_string, )
+from tests.helper_functions import score_model_in_sagemaker_docker_container
+
+# TODO you must update this
+license_keys = {
+    "AWS_ACCESS_KEY_ID": None,
+    "AWS_SECRET_ACCESS_KEY": None,
+    "SPARK_NLP_LICENSE": None,
+    "SECRET": None}
+
+os.environ.update(license_keys)
+
+# Before running any tests make sure the following environment variables are set:
+
+if not license_keys['SECRET']:
+    raise ValueError("""
+Please set SECRET, AWS_ACCESS_KEY_ID ,AWS_SECRET_ACCESS_KEY ,SPARK_NLP_LICENSE in your environment variables
+""")
+
+# nlu_model = 'tokenize' # You can use this alternatively for a leightweight test run with a tokenizer model
+nlu_model = 'en.classify.bert_sequence.covid_sentiment'
+
+
+def load_and_init_model(model=nlu_model):
+    nlp.start(model_cache_folder='/home/ckl/dump/cache_pretrained')
+    jsl_model = nlp.load(model, verbose=False)
+    return jsl_model
+
+
+def fix_dataframe_with_respect_for_nlu_issues(d1, d2):
```

Comment:
please make this a fixture

Reply:
This does not work as a fixture I think, since it's called multiple times dynamically on different inputs. 
It's doing some postprocessing for the assertions.